base/compiler/typeinfer.jl

```diff
@@ -300,9 +292,10 @@ function CodeInstance(
     local const_flags::Int32
     result_type = result.result
     @assert !(result_type isa LimitedAccuracy)
-    if inferred_result isa ConstAPI
+
+    if isa(result_type, Const) && is_total(result.ipo_effects)
```

Comment:
I think we only require `is_foldable && is_nothrow` here?

Reply:
And I think we should also check `is_inlineable_constant(result_type.value)` here.

Reply:
> I think we only require is_foldable && is_nothrow here?

I think `is_total()` is defined as ` is_foldable` && `is_nothrow`?

Reply:
> I think `is_total()` is defined as `is_foldable` && `is_nothrow`?

Yes, I just remembered it. Perhaps we should have revised its definition as we added the additional effects. "is total" is now a much stronger condition that covers the other guarantees, such as `is_notaskstate` and `is_inaccessiblememonly`. So be safe to use `is_foldable && is_nothrow` here and in the inlining handling.

Reply:
So you're saying that `is_total()` may change its definition in the future?